comid/integrityregisters.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package comid
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/veraison/swid"
+)
+
+const TextType = "text"
+
+// IRegisterIndex is the interface to hold register index
+type IRegisterIndex interface{}
+
+// IntegrityRegisters holds the Integrity Registers
+type IntegrityRegisters struct {
+	m map[IRegisterIndex]Digests
+}
+
+func NewIntegrityRegisters() *IntegrityRegisters {
+	return &IntegrityRegisters{m: make(map[IRegisterIndex]Digests)}
+}
+
+// AddDigests allows inserting a series of Digests at a specific RegisterIndex
+func (i *IntegrityRegisters) AddDigests(index IRegisterIndex, digests Digests) error {
+	if len(digests) == 0 {
+		return fmt.Errorf("no digests to add")
+	}
+	for _, digest := range digests {
+		if err := i.AddDigest(index, digest); err != nil {
+			return fmt.Errorf("unable to add Digest: %w", err)
+		}
+	}
+	return nil
+}
+
+// AddDigest allows inserting a Digest at a specific registerIndex
```

Comment:
use the actual parameter names in the docstring

```suggestion
// AddDigest allows inserting a digest at a specific index
```

The docstring should also say what kinds of index types are supported.

Reply:
Agree and changes made